test_integration/test_ee_short_region_slop.py

```diff
@@ -0,0 +1,300 @@
+            launcher_fixed.set_num_node(num_node)
+            launcher_fixed.set_num_rank(num_rank)
+            # Run the test application
+            launcher_fixed.run(test_name)
+
+            # Create the test launcher with the above configuration
+            launcher_dynamic = geopm_test_launcher.TestLauncher(app_conf,
+                                                                agent_conf_dynamic,
+                                                                cls._report_path_dynamic,
+                                                                time_limit=time_limit)
+            launcher_dynamic.set_num_node(num_node)
+            launcher_dynamic.set_num_rank(num_rank)
+            # Run the test application
+            launcher_dynamic.run(test_name)
+
+    @classmethod
+    def tearDownClass(cls):
+        """Clean up any files that may have been created during the test if we
+        are not handling an exception and the GEOPM_KEEP_FILES
+        environment variable is unset.
+
+        """
+
+        if (sys.exc_info() == (None, None, None) and not
+            cls._keep_files and not
+            cls._skip_launch):
+            os.unlink(cls._agent_conf_fixed_path)
+            os.unlink(cls._agent_conf_dynamic_path)
+            os.unlink(cls._report_path_fixed)
+            os.unlink(cls._report_path_dynamic)
+            os.unlink(cls._image_path)
+
+    def test_generate_plot(self):
+        """Visualize the data in the report
+
+        """
+        report_dynamic = geopmpy.io.RawReport(self._report_path_dynamic)
+        cols, scaling_data_dynamic, timed_data_dynamic = extract_data(report_dynamic)
+
+        report_fixed = geopmpy.io.RawReport(self._report_path_fixed)
+        cols, scaling_data_fixed, timed_data_fixed = extract_data(report_fixed)
+
+        all_energy = list(scaling_data_fixed[column_idx(cols, 'package-energy')])
+        all_energy.extend(scaling_data_dynamic[column_idx(cols, 'package-energy')])
+        all_energy.extend(timed_data_fixed[column_idx(cols, 'package-energy')])
+        all_energy.extend(timed_data_dynamic[column_idx(cols, 'package-energy')])
+        min_energy = min(all_energy)
+        max_energy = max(all_energy)
+        delta_energy = max_energy - min_energy
+        min_energy -= 0.05 * delta_energy
+        max_energy += 0.05 * delta_energy
+
+        all_runtime = list(scaling_data_fixed[column_idx(cols, 'runtime')])
+        all_runtime.extend(scaling_data_dynamic[column_idx(cols, 'runtime')])
+        all_runtime.extend(timed_data_fixed[column_idx(cols, 'runtime')])
+        all_runtime.extend(timed_data_dynamic[column_idx(cols, 'runtime')])
+        min_runtime = min(all_runtime)
+        max_runtime = max(all_runtime)
+        delta_runtime = max_runtime - min_runtime
+        min_runtime -= 0.05 * delta_runtime
+        max_runtime += 0.05 * delta_runtime
+
+        plt.figure(figsize=(11,11))
```

Comment:
This code that uses global plt directly could instead use the objects returned by plt.subplots()

Reply:
Maybe next time?